test/e2e/user_workloads_test.go

```diff
@@ -38,73 +37,65 @@ func (s *userWorkloadsTestSuite) TestIdlerAndPriorityClass() {
-	_, err = memberAwait.WaitForPods(idlerNoise.Name, len(podsNoise), wait.PodRunning(), wait.WithPodLabel("idler", "idler"), wait.WithSandboxPriorityClass())
-	require.NoError(s.T(), err)
-	_, err = memberAwait.WaitForPods("workloads-noise", len(externalNsPodsNoise), wait.PodRunning(), wait.WithPodLabel("idler", "idler"), wait.WithOriginalPriorityClass())
+	memberAwait.WaitForPods(s.T(), idlerNoise.Name, len(podsNoise), wait.PodRunning(), wait.WithPodLabel("idler", "idler"), wait.WithSandboxPriorityClass())
+	memberAwait.WaitForPods(s.T(), "workloads-noise", len(externalNsPodsNoise), wait.PodRunning(), wait.WithPodLabel("idler", "idler"), wait.WithOriginalPriorityClass())
 	require.NoError(s.T(), err)
-	_, err = hostAwait.WithRetryOptions(wait.TimeoutOption(10*time.Second)).WaitForNotificationWithName("test-idler-stage-idled", toolchainv1alpha1.NotificationTypeIdled, wait.UntilNotificationHasConditions(Sent()))
-	require.True(s.T(), errors.IsNotFound(err))
+	hostAwait.WithRetryOptions(wait.TimeoutOption(10*time.Second)).WaitForNotificationWithName(s.T(), "test-idler-stage-idled", toolchainv1alpha1.NotificationTypeIdled, wait.UntilNotificationHasConditions(Sent()))
 
 	// Check if notification has been deleted before creating another pod
-	err = hostAwait.WaitUntilNotificationWithNameDeleted("test-idler-dev-idled")
-	require.NoError(s.T(), err)
+	hostAwait.WaitUntilNotificationWithNameDeleted(s.T(), "test-idler-dev-idled")
 
 	// Create another pod and make sure it's deleted.
 	// In the tests above the Idler reconcile was triggered after we changed the Idler resource (to set a short timeout).
 	// Now we want to verify that the idler reconcile is triggered without modifying the Idler resource.
 	// Notification shouldn't be created again.
-	pod := s.createStandalonePod(idler.Name, "idler-test-pod-2")    // create just one standalone pod. No need to create all possible pod controllers which may own pods.
-	_, err = memberAwait.WaitForPod(idler.Name, "idler-test-pod-2") // pod was created
+	pod := s.createStandalonePod(idler.Name, "idler-test-pod-2")  // create just one standalone pod. No need to create all possible pod controllers which may own pods.
+	memberAwait.WaitForPod(s.T(), idler.Name, "idler-test-pod-2") // pod was created
 	require.NoError(s.T(), err)
 	time.Sleep(time.Duration(2*idler.Spec.TimeoutSeconds) * time.Second)
-	err = memberAwait.WaitUntilPodDeleted(pod.Namespace, pod.Name)
-	require.NoError(s.T(), err)
-	_, err = hostAwait.WithRetryOptions(wait.TimeoutOption(10*time.Second)).WaitForNotificationWithName("test-idler-dev-idled", toolchainv1alpha1.NotificationTypeIdled, wait.UntilNotificationHasConditions(Sent()))
-	require.True(s.T(), errors.IsNotFound(err))
+	memberAwait.WaitUntilPodDeleted(s.T(), pod.Namespace, pod.Name)
+	hostAwait.WithRetryOptions(wait.TimeoutOption(10*time.Second)).WaitForNotificationWithName(s.T(), "test-idler-dev-idled", toolchainv1alpha1.NotificationTypeIdled, wait.UntilNotificationHasConditions(Sent()))
```

Comment:
In this case we expected to receive an error. The line that was removed below was `require.True(s.T(), errors.IsNotFound(err))` but `WaitForNotificationWithName()` now has `require.NoError(t, err)` as part of the function so I'd expect this test is failing now? I think the previous pattern of calling functions and then validating the returned error is more flexible because sometimes we do expect an error and that's a legitimate test. Might be better to go back to that pattern. WDYT?

Reply:
I still prefer checking the errors within the `WaitFor...` funcs. But for that particular case, we could have a new `CheckNotificationWithNameNotFound` which would assert that `errors.IsNotFound(err)`. WDYT?

Reply:
well... after discussing with @MatousJobanek we agreed that returning an `(object, error)` is probably more idiomatic, so I'll revert my changes ;)